src/savings/savings.ts

```diff
@@ -0,0 +1,24 @@
+import { Provider, Signer, ethers } from 'ethers';
+import { Decimal } from '@tempusfinance/decimal';
+import { RaftConfig } from '../config';
+import { RSavingsModule, RSavingsModule__factory } from '../typechain';
+
+export class Savings {
+  protected providerOrSigner: Provider | Signer;
+  protected rSavingsModuleContract: RSavingsModule;
+
+  constructor(providerOrSigner: Provider | Signer) {
```

Comment:
```suggestion
  constructor(providerOrSigner: ContractRunner) {
```
`ContractRunner` can be used as an abstract type instead.